### PR TITLE
ddl: modify `cancel drop column` ddl test

### PR DESCRIPTION
**Pull request by ciscoxll**
### What problem does this PR solve? 
From https://github.com/pingcap/tidb/pull/8545 `TestCancelJob` the test here is invalid.

### What is changed and how it works?

In the test of `cancel drop column`, if column does not exist, you need to add column first, cancel job and check it separately. `StateDeleteOnly` and `StateWriteOnly` and `StateWriteReorganization` are in compliance with expectations.

### Check List 

Tests 

 - Unit test

---
This change is [Reviewable](https://reviewable.io/reviews/pingcap/tidb/8897)

### CODE REVIEW

**Comment by ciscoxll** [comment]
@winkyao @zimulala PTAL

**Comment by ciscoxll** [comment]
@crazycs520 PTAL

File: ddl/ddl_test.go [hunk 1]
```diff
@@ -200,6 +200,21 @@ func testCreateIndex(c *C, ctx sessionctx.Context, d *ddl, dbInfo *model.DBInfo,
 	return job
 }
 
+func testCancelDropColumn(c *C, ctx sessionctx.Context, d *ddl, dbInfo *model.DBInfo, tblInfo *model.TableInfo, args []interface{}) *model.Job {
```
Comment (zimulala) [inline ddl/ddl_test.go]: Could we use `testDropColumn` instead？

Comment (crazycs520) [reply ddl/ddl_test.go]: @ciscoxll Please address the comment.

Comment (ciscoxll) [reply ddl/ddl_test.go]: @zimulala Done

**Comment by ciscoxll** [comment]
@zimulala @winkyao PTAL

**Comment by ciscoxll** [comment]
@zimulala @winkyao PTAL

**Pull request merged**